pkg/daemon/controller/controller.go

```diff
@@ -352,6 +354,31 @@ func (sc *ServiceController) expectedServiceCommand(expected servicescm.Service)
 	for key, value := range psVars {
 		expectedCmd = strings.ReplaceAll(expectedCmd, key, value)
 	}
+	// TODO: This goes against WICD design principles and needs to be changed https://issues.redhat.com/browse/WINC-896
+	//       WICD should not have special casing like this, and should use the services ConfigMap as its source of truth
+	if strings.Contains(expectedCmd, services.NodeIPVar) {
+		// Set NodeIP to IPv4 value by matching this instance's addresses to those in the Windows instances ConfigMap
+		instances, err := wiparser.GetInstances(sc.client, sc.watchNamespace)
+		if err != nil {
+			return "", err
+		}
+		addrs, err := LocalInterfaceAddresses()
+		if err != nil {
+			return "", err
+		}
+		nodeIPValue := ""
+		for _, addr := range addrs {
+			for _, instance := range instances {
+				if instance.IPv4Address == addr.String() {
```

Comment:
Shouldn't we also compare the address listed in the instances ConfigMap instead? Or am I missing something that makes this check unnecessary

`if instance.Address == addr.String() || instance.IPv4Address == addr.String() {`

Reply:
I suspect this is why the platform-none test is failing with `unable to find IPv4 address to use as Node IP for this instance` -- there is no guarantee the interface addresses for a specific interface will contain an IPv4 address, right?

Reply:
It looks like your latest debug commit makes it clear. `addr.String()` returns a value with subnet appended

```
I0928 22:14:47.569874    4660 controller.go:372] Checking if 192.168.59.27 == fe80::f127:441b:58:4b46/64
I0928 22:14:47.569874    4660 controller.go:372] Checking if 192.168.59.27 == 192.168.59.27/27
I0928 22:14:47.569874    4660 controller.go:372] Checking if 192.168.59.27 == ::1/128
I0928 22:14:47.569874    4660 controller.go:372] Checking if 192.168.59.27 == 127.0.0.1/8
I0928 22:14:47.569874    4660 controller.go:372] Checking if 192.168.59.27 == fe80::d156:975f:69ef:f1c4/64
I0928 22:14:47.569874    4660 controller.go:372] Checking if 192.168.59.27 == 172.27.0.1/20
E0928 22:14:47.569874    4660 controller.go:252] unable to find IPv4 address to use as Node IP for this instance
```

Reply:
The test was failing because the local address wasn't being converted into an ipv4 address before the comparison.
String-ing a net.Addr will result in an address like `192.168.0.1/32 `, stringing a net.IP will give `192.168.0.1`

Reply:
> there is no guarantee the list of all interface addresses on a specific instance will contain an IPv4 address, right?

As part of `GetInstances()` .IPv4Address needs to be populated or else the CM parse will fail.

Reply:
I guess my question is related to the instance -- will the addr list returned by `LocalInterfaceAddresses()` always contain an IPv4 value? If not, then IMO we should also compare with `instance.Address`

Reply:
It would as per the requirements of WMCO. We don't support IPv6.